src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseService.java

```diff
@@ -777,4 +777,16 @@ public long countSubmissionsByCourseIdSubmitted(Long courseId) {
         log.debug("countSubmissionsByCourseIdSubmitted took " + (System.currentTimeMillis() - start) + "ms");
         return count;
     }
+
+    /**
+     * Sets the transient attribute "testRepositoryUrl" if the exercises is a programming exercise
+     *
+     * @param exercise the exercise for which to set the url
+     */
+    public void setTestRepositoryUrlForProgrammingExercise(Exercise exercise) {
+        if (exercise instanceof ProgrammingExercise) {
+            boolean isLocalSimulation = (((ProgrammingExercise) exercise).getTestRepositoryUrl()).contains("artemislocalhost");
+            ((ProgrammingExercise) exercise).setIsLocalSimulation(isLocalSimulation);
```

Comment:
I would only set this value if it is a local simulation

Reply:
now the value will be only set if it is a local simulation